Source/Scene/Points3DTileContent.js

```diff
@@ -340,12 +423,80 @@ define([
         var positions = parsedContent.positions;
         var colors = parsedContent.colors;
         var normals = parsedContent.normals;
+        var batchIds = parsedContent.batchIds;
         var isQuantized = parsedContent.isQuantized;
         var isOctEncoded16P = parsedContent.isOctEncoded16P;
+        var styleableProperties = parsedContent.styleableProperties;
         var isTranslucent = content._isTranslucent;
 
         var hasColors = defined(colors);
         var hasNormals = defined(normals);
+        var hasBatchIds = defined(batchIds);
+
+        var batchTable = content.batchTable;
+        var hasBatchTable = defined(batchTable);
+        var hasStyleableProperties = defined(styleableProperties);
+
+        // Use per-point normals to hide back-facing points.
+        var backFaceCulling = true;
+
+        var positionAttributeLocation = 0;
+        var colorAttributeLocation = 1;
+        var normalAttributeLocation = 2;
+        var batchIdAttributeLocation = 3;
+        var numberOfAttributes = 4;
+
+        var styleableShaderAttributes;
+        var styleableVertexAttributes;
+        var styleableVertexAttributeLocations;
+
+        if (hasStyleableProperties) {
+            styleableShaderAttributes = '';
+            styleableVertexAttributes = [];
+            styleableVertexAttributeLocations = {};
+
+            var attributeLocation = numberOfAttributes;
+
+            for (var name in styleableProperties) {
+                if (styleableProperties.hasOwnProperty(name)) {
+                    var property = styleableProperties[name];
+                    // TODO : this will not handle matrix types currently
+                    var componentCount = property.componentCount;
+                    var typedArray = property.typedArray;
+                    var componentDatatype = ComponentDatatype.fromTypedArray(typedArray);
+
+                    // Append attributes to shader
+                    var attributeName = 'style_' + name;
+                    var attributeType;
+                    if (componentCount === 1) {
+                        attributeType = 'float';
+                    } else {
+                        attributeType = 'vec' + componentCount;
+                    }
+                    styleableShaderAttributes += 'attribute ' + attributeType + ' ' + attributeName + '; \n';
+
+                    var vertexBuffer = Buffer.createVertexBuffer({
+                        context : context,
+                        typedArray : property.typedArray,
+                        usage : BufferUsage.STATIC_DRAW
+                    });
+
+                    var vertexAttribute = {
+                        index : attributeLocation,
+                        vertexBuffer : vertexBuffer,
+                        componentsPerAttribute : componentCount,
+                        componentDatatype : componentDatatype,
+                        normalize : false, // TODO : support this in batch table binary spec?
```

Comment:
Do you think we can go without it?  Do you have another killer use case other than unsigned byte RGBA, which is already in the feature table?